Creating-Interactions.md

```diff
@@ -1,62 +1,158 @@
+     * `my-input-interaction.component.html`
+     * `oppia-interactive-my-input.component.ts`: Defines a `InteractiveMyInputComponent` class.
 
-Once you've created these files, activate your interaction in Oppia by editing the `ALLOWED_INTERACTION_CATEGORIES` variable in `feconf.py` (you'll need to specify its name). Interactions are classified under different categories, add yours to a suitable category. Your interaction should then be available in the interaction repository, and can be used in explorations.
+     * `oppia-interactive-my-input.component.spec.ts`
 
-### Testing ###
+   * Response component:
+
+     * `my-input-response.component.html`
+     * `oppia-response-my-input.component.ts`: Defines a `ResponseMyInputComponent` class.
+     * `oppia-response-my-input.component.spec.ts`
+
+   * Short response component:
+
+     * `my-input-short-response.component.html`
+     * `oppia-short-response-my-input.component.ts`: Defines a `ShortResponseMyInputComponent` class.
+     * `oppia-short-response-my-input.component.spec.ts`
+
+   * Rules service:
+
+     * `my-input-rules.service.ts`
+     * `my-input-rules.service.spec.ts`
+
+   * Validation service:
+
+     * `my-input-validation.service.ts`
+     * `my-input-validation.service.spec.ts`
+
+   Each component should implement `OnInit`. Parameters from the Python file are accessible as `parameterWithValue` where the Python variable is `parameter`. The `WithValue` is a legacy hold-over. Note that the parameters in $attrs are JSON-stringified, so you will need to use `oppiaHtmlEscaper.escapedJsonToObj()` to convert them back. (Do not use `JSON.parse()`, since it does not handle the necessary escaping.)
+
+7. Create a `MyInput/MyInput.ts` file that uses `require()` or `import` statements to import all the `*.ts` files in `MyInput/directives`. If you have CSS files in your `static/` folder, import those here too.
+
+8. Create a `MyInput/my-interaction-interactions.module.ts` file that imports the interaction, response, and short response components from `MyInput/directives`. It should also import any dependencies and export a module like this:
+
+   ```ts
+   @NgModule({
+     imports: [
+       ... dependencies ...
+     ],
+     declarations: [
+       ... interaction, response, and short response components ...
+     ],
+     entryComponents: [
+       ... interaction, response, and short response components ...
+     ],
+     exports: [
+       ... interaction, response, and short response components ...
+     ],
+   })
+
+   export class MyInputInteractionModule { }
+   ```
+
+9. Activate your interaction in Oppia by editing the `ALLOWED_INTERACTION_CATEGORIES` variable in `assets/constants.ts` (you'll need to specify its name). Interactions are classified under different categories, add yours to a suitable category. Your interaction should then be available in the interaction repository, and can be used in explorations.
```

Comment:
The comma after "categories" should be a semicolon.

Reply:
done